src/utils/string.ts

```diff
@@ -0,0 +1,144 @@
+  if (pattern.includes('(') && pattern.includes(')')) {
+    // Defined regex to extract url from the pattern defined in the configuration file
+    // Example: [{{{ Number of records }}}](/deriva-webapps/plot/?config=gudmap-todate-pie){target=_blank}
+    // extractedLink = /deriva-webapps/plot/?config=gudmap-todate-pie
+    // Extracts all the characters placed between "( )".
+    // "]/(" : find a ']' bracket followed by '('
+    // "." : matches any character and
+    // "*?" : matches the previous token between zero and unlimited times
+    // "i" modifier :  insensitive. Case insensitive match (ignores case of [a-zA-Z])
+    // "g" modifier :  global. All matches.
+    const markdownUrlRegex = /]\((.*?)\)/gi;
+    match = markdownUrlRegex.exec(pattern);
+    extractedLink = match ? match[1] : false;
+  } else if (pattern.includes('href')) {
+    // Defined regex to extract url from the generated html element with href attribute
+    // Example: <a href="(/deriva-webapps/plot/?config=gudmap-todate-pie" target="_blank">prostate gland</a>
+    // extractedLink = /deriva-webapps/plot/?config=gudmap-todate-pie
+    // Extracts a link from the anchor tag
+    // "\s" : matches a space character
+    // ^\n\r : matches a string that does not have new line or carriage return
+    const htmlUrlRegex = /<a\shref="([^\n\r]*?)"/gi;
+    match = htmlUrlRegex.exec(pattern);
+    extractedLink = match ? match[1] : false;
+  }
+
+  // return false if no extracted link
+  return extractedLink;
+};
+
+/**
+ * Formats the string data used by plot by adding commas for numbers when neccesary
+ *
+ * @param data data value
+ * @param format whether to format or not
+ * @param type the type of plot data
+ * @returns
+ */
+export const formatPlotData = (data: string, format: boolean, type: string): string | number => {
+  if (format) {
+    try {
+      const formatedData = parseInt(data.split(' ')[0], 10);
+      if (isNaN(formatedData)) {
+        return data;
+      }
+      if (type === 'pie') {
+        return formatedData.toString();
+      }
+      // this regex is used to add a thousand separator in the number if possible
+      return addComma(formatedData);
+    } catch (e) {
+      return data;
+    }
+  } else {
+    if (type === 'pie' || typeof data !== 'number') {
+      return data;
+    }
+    return addComma(data);
+  }
+};
+
+/**
+ * Adds a thousand separator in the given number if possible
+ *
+ * @param data data to add thousand separator
+ * @returns
+ */
+export const addComma = (data: number | string): string => {
+  return data.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,');
+};
+
+/**
+ * Creates the link bemplateased on the given markdown pattern and templateParam
+ *
+ * @param markdownPattern pattern used to create link 
+ * @param templateParam params used to replace values in the pattern to create the link
+ * @returns
+ */
+export const createLink = (markdownPattern: string, templateParam?: any): string => {
```

Comment:
Created some helper functions that were often getting used in plot.app.js

@jrchudy